src/Led.cpp

```diff
@@ -224,500 +255,566 @@ static void Led_Task(void *parameter) {
+					// Animation of Error or OK
+					// --------------------------------------------------
+					case LedAnimationType::Error: 
+					case LedAnimationType::Ok:{
+						CRGB signalColor = CRGB::Green;
+						uint16_t onTime = 400;
+						animationActive = true;
+						if(activeAnnimation == LedAnimationType::Error) {
+							signalColor = CRGB::Red;
+							onTime = 200;
+						}	
+
+						if (NUM_LEDS == 1) {
+							FastLED.clear();
+							if (singleLedStatus) {
+								leds[0] = signalColor;
 							}
 							FastLED.show();
-							vTaskDelay(intervalToLongPress / NUM_LEDS * portTICK_RATE_MS);
-						}
-					}
-				}
-			} else {
-				gShutdownButton = (sizeof(gButtons) / sizeof(gButtons[0])) - 1; // If CMD_SLEEPMODE was not assigned to an enabled button, dummy-button is used
-				if (!gButtons[gShutdownButton].currentState) {
-					gButtons[gShutdownButton].currentState = true;
-				}
-			}
-
-			// Multi-LED: all leds flash red 1x
-			// Single-LED: led flashes red 5x
-			if (LED_INDICATOR_IS_SET(LedIndicatorType::Error)) { // If error occured (e.g. RFID-modification not accepted)
-				LED_INDICATOR_CLEAR(LedIndicatorType::Error);
-				requestProgressRedraw = true;
-				FastLED.clear();
+							singleLedStatus = !singleLedStatus;
 
-				if (NUM_LEDS == 1) {
-					for (uint8_t cnt = 0; cnt < 5; cnt++) {
-						FastLED.clear();
-						if (singleLedStatus) {
-							leds[0] = CRGB::Red;
+							if (animationIndex < 5){
+								animationIndex++;
+								animaitonTimer = 100;
+							} else {
+								animationActive = false;
+							}
 						} else {
-							leds[0] = CRGB::Black;
+							fill_solid(leds, NUM_LEDS, signalColor);
+							FastLED.show();
+							if (animationIndex > 0){
+								animationActive = false;
+							} else {
+								animationIndex++;
+								animaitonTimer = onTime;
+							}
 						}
-						FastLED.show();
-						singleLedStatus = !singleLedStatus;
-						vTaskDelay(portTICK_RATE_MS * 100);
-					}
-				} else {
-					for (uint8_t led = 0; led < NUM_LEDS; led++) {
-						leds[Led_Address(led)] = CRGB::Red;
-					}
-					FastLED.show();
-					vTaskDelay(portTICK_RATE_MS * 200);
-				}
-			}
+					} break;
+
+					// --------------------------------------------------
+					// Animation of Volume
+					// --------------------------------------------------
+					case LedAnimationType::Volume: {
+						/*
+						* - Single-LED: led indicates loudness between green (low) => red (high)
+						* - Multiple-LEDs: number of LEDs indicate loudness; gradient is shown between
+						*   green (low) => red (high)
+						*/
+						animationActive = true;
+
+						// wait for further volume changes within next 20ms for 50 cycles = 1s
+						uint32_t ledValue =  map(AudioPlayer_GetCurrentVolume(), 0,
+										AudioPlayer_GetMaxVolume(), 0,
+										NUM_LEDS * DIMMABLE_STATES);
+						uint8_t fullLeds = ledValue/DIMMABLE_STATES;
+						uint8_t lastLed = ledValue%DIMMABLE_STATES;
```

Comment:
Overlooked this one yesterday. Also const candidates

```suggestion
						const uint32_t ledValue =  map(AudioPlayer_GetCurrentVolume(), 0,
										AudioPlayer_GetMaxVolume(), 0,
										NUM_LEDS * DIMMABLE_STATES);
						const uint8_t fullLeds = ledValue/DIMMABLE_STATES;
						const uint8_t lastLed = ledValue%DIMMABLE_STATES;
```